relay/sources/relayd/src/data/reporting.rs

```diff
@@ -129,40 +126,48 @@ named!(parse_runlog<CompleteStr, Vec<RawReport>>,
     )
 );
 
+fn parse_date(input: CompleteStr) -> Result<DateTime<FixedOffset>, chrono::format::ParseError> {
+    DateTime::parse_from_str(input.as_ref(), "%Y-%m-%d %H:%M:%S%z")
+}
+
+fn parse_i32(input: CompleteStr) -> IResult<CompleteStr, i32> {
+    parse_to!(input, i32)
+}
+
 named!(report<CompleteStr, RawReport>, do_parse!(
     // TODO NOT CORRECT
-    // pas de line break dans une ligne
-    // gérer les reports interrompus
+    // no line break inside a filed (except message)
+    // handle partial reports without breaking following ones
     logs: log_entries >>
     rudder_report_begin >>
     policy: take_until_and_consume_s!("@@") >>
     event_type: take_until_and_consume_s!("@@") >>
     rule_id: take_until_and_consume_s!("@@") >>
     directive_id: take_until_and_consume_s!("@@") >>
-    serial: take_until_and_consume_s!("@@") >>
+    serial: map_res!(take_until_and_consume_s!("@@"), parse_i32) >>
```

Comment:
serial should always be 0

Reply:
meaning: there is no way the serial can be set to something else than 0 in Rudder 4.3 and more

Reply:
The idea is to allow reusing this field if we want it in the future, without having to upgrade all relays.

Reply:
(specially as it is cheap and just as simple to implement)